### ISSUE TITLE
It is not accessible to JAWS 16 with Internet Explorer 11

### ISSUE THREAD

**Issue by antoninadert** [sweetalert2/sweetalert2#980]
Using any of the sweetalerts in their most simple format on IE11 + JAWS 16 (version I tested with)

I will take the simplest alert as an example:

![image](https://user-images.githubusercontent.com/8929314/36907650-7598bfc8-1e39-11e8-99df-3b454b866699.png)

First problem: if the user presses up arrow, or open a list of elements, they can still move the focus out of the sweet alert. I believe it should be locked. 

Usual technique for this problem is to add `aria-hidden="true"` to the whole document (except the content inside the sweetalert)

Second problem is the way thing is read: It first reads the title (normal) then the button, but it skipped the "You clicked the button!" part

then it reads everything again (not ideal but it works) 

Third  problemt is that "You clicked the button!" part is actually a checkbox regarding what is announced by JAWS. It should be a simple text instead.

Is there anyone else testing Accessibility or am I the first one ? 

By the way, thanks for this awesome project

**Comment by limonte** [comment]
Hello @antoninadert and thanks a lot for reporting JAWS/IE11 issues!

According to the (latest WebAIM Screen Reader User Survey)[https://webaim.org/projects/screenreadersurvey7/) JAWS is the most popular screen reader (46.6%) and we definitely need to support it with all browsers including IE11.

I'll try to fix all reported issues this week. If you have some clue about what could be the cause and solution for the 2nd and the 3rd issues please let me know - currentry I have no idea why JAWS read the modal content incorrectly.

**Comment by antoninadert** [comment]
I don't have a lot of time but here is my quick try:
```
<div class="swal2-container swal2-center swal2-fade swal2-shown" style="-ms-overflow-y: auto;">
  <div class="swal2-popup swal2-modal swal2-show" style="display: flex;">
    <div class="swal2-header">
      <ul class="swal2-progresssteps" style="display: none;"></ul>
      <div class="swal2-icon swal2-error" style="display: none;">
        <span class="swal2-x-mark">
          <span class="swal2-x-mark-line-left"></span>
          <span class="swal2-x-mark-line-right"></span>
        </span>
      </div>
      <div class="swal2-icon swal2-question" style="display: none;">
        <span class="swal2-icon-text">?</span>
      </div>
      <div class="swal2-icon swal2-warning" style="display: none;">
        <span class="swal2-icon-text">!</span>
      </div>
      <div class="swal2-icon swal2-info" style="display: none;">
        <span class="swal2-icon-text">i</span>
      </div>
      <div class="swal2-icon swal2-success swal2-animate-success-icon" style="display: flex;">
        <div class="swal2-success-circular-line-left" style="background-color: rgb(255, 255, 255);"></div>
        <span class="swal2-success-line-tip"></span>
        <span class="swal2-success-line-long"></span>
        <div class="swal2-success-ring"></div>
        <div class="swal2-success-fix" style="background-color: rgb(255, 255, 255);"></div>
        <div class="swal2-success-circular-line-right" style="background-color: rgb(255, 255, 255);"></div>
      </div>
      <img class="swal2-image" style="display: none;">
      <h2 class="swal2-title" id="swal2-title">Good job!</h2>
      <button class="swal2-close" style="display: none;" type="button">×</button>
    </div>
    <div class="swal2-content">
      <div>You clicked the button!</div>
    </div>
    <div class="swal2-actions" style="display: flex;">
      <button class="swal2-confirm swal2-styled" style="border-right-color: rgb(48, 133, 214); border-left-color: rgb(48, 133, 214);"
        aria-label="" type="button">OK</button>
      <button class="swal2-cancel swal2-styled" style="display: none;" aria-label="" type="button">Cancel</button>
    </div>
    <div class="swal2-footer" style="display: none;"></div>
  </div>
</div>
```

I removed a lot of stuff regarding the content (stuff that was in the DOM) and apparently it doesn't say 'clickable' anymore. I don't know what caused this 'clickable' thing to be read, so you might try to add things back incrementally until it breaks.

 (sadly I didn't have a codepen so I tried with the inspector, but with IE I suspect the DOM does not update with JAWS in a consistent manner, so it may be tricky)

**Comment by antoninadert** [comment]
Well I did another test: 
![image](https://user-images.githubusercontent.com/8929314/36970106-b11e1978-2067-11e8-9f71-7d2ad1a96b35.png)
I added a random DIV below the first swal container, and now it doesn't read the 'clickable' in the 'you clicked the button' label below.

The weird thing is that if I remove this  'LOL' label from the inspector, it doesn't read clickable afterwards...

No clue why

**Comment by limonte** [comment]
@antoninadert apologies for the delayed response! I will dedicate the timebox on Saturday for a11y since we have the issue with NVDA as well https://github.com/sweetalert2/sweetalert2/issues/1010

**Comment by antoninadert** [comment]
The NVDA issue probably covers the fact you have to use aria-hidden on the whole HTML except the div of the modal. 
Other issues remain here, but probably are shared with NVDA as well.

**Comment by antoninadert** [comment]
Hello everyone as issue #1010 has been closed, what is the progress done on this one?

**Comment by limonte** [comment]
Hello @antoninadert and sorry for the delayed response. I happy to announce that the issue with focus has been fixed!

As to the second and the third issues, I'm not able to reproduce them with the latest JAWS + IE11.

Could you please verify the first issue and tell if second and third are still happening in your environment or not?

Thanks in advance!

**Comment by antoninadert** [comment]
Hello @limonte I tried the first sweetalert on your official website and it correctly locked the focus inside the modal (very good !)

I tested with JAWS 16.0 and IE 11

Now I will tell you what it reads and where is the **focus** (in bold):

At opening, what is read automatically without user action: 

> * "Good job heading level 2. Clickable"
> * "OK Button"

At pressing down arrow:

> * "OK Button" 

: **OK button** took focus

By going from top to bottom:

> * "Good job heading level 2. Clickable"
> * "You clicked the button. Clickable"
> * "Checkbox not checked
> * "OK Button"

-------
Now: What I would expect to be read:
--------

At opening, what is read automatically without user action: 

> * "Good job heading level 2"
> * "You clicked the button.
> * "OK Button"

At pressing down arrow:

> * "You clicked the button. 
> * "OK Button" 

: **Good Job (H2)** should take focus


By going from top to bottom:

> * "Good job heading level 2"
> * "You clicked the button"
> * "OK Button"

**Comment by limonte** [comment]
Hello @antoninadert and thank you for your prompt feedback!

JAWS 16.0 seems to be outdated, could you please upgrade JAWS to the latest version and try again.

![image](https://user-images.githubusercontent.com/6059356/43388748-0ec7ef14-943e-11e8-862a-9d07c0104f20.png)

**Comment by antoninadert** [comment]
Sadly I am not able to perform an update as of now...

**Comment by limonte** [comment]
> Sadly I am not able to perform an update as of now...

Hm, I wonder why? Is that a common thing among JAWS users that they stick to older versions and unable to update?

**Comment by antoninadert** [comment]
You need to pay all major updates.

**Comment by limonte** [comment]
> You need to pay all major updates.

Didn't know about that, thanks for the info! 

---

JAWS 16 hasn't been updated more than 1 year and I don't think it's a good idea to spend time and add hacks/workarounds for old screen readers to SweetAlert2's codebase.

I value the code maintainability and bundle size more than old screen readers support.

---

It seems that the main issue with the focus lock has been fixed and others are JAWS 16 specific. JAWS 16 won't be supported, I'm closing this issue.

Thanks a lot for your time and valuable feedback @antoninadert!
